tools/toolchain/optimized_clang.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash -uex
+
+if [[ -z "${CLANG_BUILD}" ]]; then
+    echo "CLANG_BUILD not specified"
+    exit 1
+elif [[ "${CLANG_BUILD}" = "SKIP" ]]; then
+    echo "Skip building optimized clang"
+    exit 0
+elif [[ "${CLANG_BUILD}" = "INSTALL" ]]; then
+    echo "build and install optimized clang"
+elif [[ "${CLANG_BUILD}" = "INSTALL_FROM" ]]; then
+    echo "install prebuilt optimized clang"
+else
+    echo "Not sure what to do with ${CLANG_BUILD}"
+    exit 1
+fi
+
+if [[ "${CLANG_BUILD}" = "INSTALL" ]] || [[ "${CLANG_BUILD}" = "INSTALL_FROM" ]]; then
+    if [[ -z "${CLANG_ARCHIVE}" ]]; then
+        echo "CLANG_ARCHIVE does not specified"
+        exit 1
+    fi
+fi
+
+ARCH="$(arch)"
+LLVM_EXPERIMENTAL_TARGETS_TO_BUILD="WebAssembly"
+if [[ "${ARCH}" = "x86_64" ]]; then
+    LLVM_TARGET_ARCH=X86
+elif [[ "${ARCH}" = "aarch64" ]]; then
+    LLVM_TARGET_ARCH=AArch64
+    LLVM_EXPERIMENTAL_TARGETS_TO_BUILD="${LLVM_EXPERIMENTAL_TARGETS_TO_BUILD};AArch64"
+else
+    echo "Unsupported architecture: ${ARCH}"
+    exit 1
+fi
+
+SCYLLA_DIR=/mnt
+CLANG_ROOT_DIR="${SCYLLA_DIR}"/clang_build
+CLANG_CHECKOUT_NAME=llvm-project
+CLANG_BUILD_DIR="${CLANG_ROOT_DIR}"/"${CLANG_CHECKOUT_NAME}"
+CLANG_BIN="${CLANG_BUILD_DIR}"/build/bin
+
+# Which LLVM release to build in order to compile Scylla
+LLVM_CLANG_TAG=16.0.6
+CLANG_SUFFIX=16
+
+CLANG_ARCHIVE=$(cd "${SCYLLA_DIR}" && realpath -m "${CLANG_ARCHIVE}")
+
+CLANG_OPTS=(-DCMAKE_C_COMPILER="/usr/bin/clang" -DCMAKE_CXX_COMPILER="/usr/bin/clang++" -DLLVM_USE_LINKER="/usr/bin/ld.lld" -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="${LLVM_TARGET_ARCH};WebAssembly" -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD="${LLVM_EXPERIMENTAL_TARGETS_TO_BUILD}" -DLLVM_TARGET_ARCH="${LLVM_TARGET_ARCH}" -G Ninja -DLLVM_INCLUDE_BENCHMARKS=OFF -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DLLVM_ENABLE_BINDINGS=OFF -DLLVM_ENABLE_PROJECTS="clang;lld" -DLLVM_ENABLE_RUNTIMES="compiler-rt" -DLLVM_ENABLE_LTO=Thin -DCLANG_DEFAULT_PIE_ON_LINUX=OFF -DLLVM_BUILD_TOOLS=OFF -DCMAKE_C_FLAGS="-Xclang -mllvm -Xclang -vp-counters-per-site=16" -DCMAKE_CXX_FLAGS="-Xclang -mllvm -Xclang -vp-counters-per-site=16")
+SCYLLA_OPTS=(--date-stamp "$(date "+%Y%m%d")" --debuginfo 1 --tests-debuginfo 1 --c-compiler="${CLANG_BIN}/clang" --compiler="${CLANG_BIN}/clang++" --build-dir="build_profile" --out="build_profile.ninja")
```

Comment:
would you please use a variable for `build_profile.ninja`? it's not a blocker though. otherwise, i would like to implement this proposal in a follow-up change.